src/lib/chat/matrix-client.ts

```diff
@@ -613,6 +613,44 @@ export class MatrixClient implements IChatClient {
     await this.matrix.redactEvent(roomId, messageId);
   }
 
+  async getRoomIdForAlias(alias: string): Promise<string | undefined> {
+    await this.waitForConnection();
+    return await this.matrix
+      .getRoomIdForAlias(alias) // { room_id, servers[] }
+      .catch((err) => {
+        if (err.errcode === 'M_NOT_FOUND') {
+          Promise.resolve(undefined);
+        }
+
+        if (err.errcode === 'M_INVALID_PARAM') {
+          // handle invalid alias error
+        }
+      })
+      .then((response) => response && response.room_id);
+  }
+
+  async apiJoinRoom(aliasOrId: string): Promise<{ success: boolean; response: any; message: string }> {
```

Comment:
I don't think this should be on the matrix-client. Ok to leave for now but it can probably just live as an independent function in the chat lib.

Reply:
i'll note it as a follow up